src/awsexplorer/awsExplorer.ts

```diff
@@ -37,60 +57,32 @@ export class AwsExplorer implements vscode.TreeDataProvider<AWSTreeNodeBase>, Re
     }
 
     public async getChildren(element?: AWSTreeNodeBase): Promise<AWSTreeNodeBase[]> {
-        if (!!element) {
-            try {
-                return await element.getChildren()
-            } catch (error) {
-                return [
-                    new AWSCommandTreeNode(
-                        element,
-                        localize(
-                            'AWS.explorerNode.lambda.retry',
-                            'Unable to load Lambda Functions, click here to retry'
-                        ),
-                        'aws.refreshAwsExplorerNode',
-                        [this, element]
-                    )
-                ]
+        let childNodes: AWSTreeNodeBase[] = []
+
+        try {
+            if (element) {
+                childNodes = childNodes.concat(await element.getChildren())
+            } else {
+                childNodes = childNodes.concat(await this.getRootNodes())
             }
-        }
+        } catch (err) {
+            const error = err as Error
+            this.logger.error(`Error getting children for node ${element?.label ?? 'Root Node'}`, error)
 
-        const profileName = this.awsContext.getCredentialProfileName()
-        if (!profileName) {
-            return [
+            childNodes.splice(
+                0,
+                childNodes.length,
                 new AWSCommandTreeNode(
-                    undefined,
-                    localize('AWS.explorerNode.signIn', 'Connect to AWS...'),
-                    'aws.login',
-                    undefined,
-                    localize('AWS.explorerNode.signIn.tooltip', 'Connect to AWS using a credential profile')
+                    element,
+                    localize('AWS.explorerNode.error.retry', 'Unable to get child nodes, click here to retry'),
+                    'aws.refreshAwsExplorerNode',
+                    [this, element],
+                    error.message
                 )
-            ]
+            )
```

Comment:
I'm not sure if I like this node on failure vs. the login node. In what cases would this fail? It feels that the only time this would be reached is if a user's credentials suddenly got invalidated, and this would essentially cause them to loop in place with the bad credentials.

Reply:
In the existing code, if an Error was thrown while producing region nodes, the Error would not be handled, leaving the Explorer in a state where it had no nodes. This change ensures all Errors are handled, producing a node so that users know that something went wrong. 

Secondary, the existing error text was lambda-centric, but the Explorer deals with more services than Lambda now.

Reply:
I'm wondering if it might be a better idea to do a toast with the error message and to reset to an unconnected state. I like the idea of the update, but I think in practice it won't be the most functional. Ultimately, I think the error message needs to be more obvious instead of just as a tooltip.

Reply:
The error isn't necessarily related to a credentials issue. This is a catch-all handling, which now covers code that previously wasn't covered. Errors _shouldn't_ be making it to this level. If they are, there was a failure on the extension developer, and this is a reasonable way for the Explorer to handle it gracefully.